runtime/observability/telemetry-api/common/src/aws/smithy/kotlin/runtime/telemetry/TelemetryConfig.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.telemetry
+
+/**
+ * Common configuration options for configuring telemetry of a component
+ */
+public interface TelemetryConfig {
+    /**
+     * The [TelemetryProvider] to instrument the component with.
+     */
+    public val telemetryProvider: TelemetryProvider
+
+    public interface Builder {
+        /**
+         * The [TelemetryProvider] to instrument the component with.
+         */
+        public var telemetryProvider: TelemetryProvider?
+    }
```

Comment:
**Question**: We intend to provide a default value of `TelemetryProvider.None` right? Why leave the `var` nullable?

Reply:
It's on the builder, this is how most of our config builders work right? `TelemetryConfig.telemetryProvider` is non nullable and will get the default.